### PR TITLE
To do list - Add/Remove

### PR DESCRIPTION
**Pull request by ReshyResh**
# Requirements fulfilled
  - Added function to add a new task.
  - Double click to edit a task / remove single item, click outside to save changes.
  - Function to clear all completed (Checked) tasks.
  - All changed are saved locally.

# Live demo
Click [here](https://reshyresh.github.io/To-Do-List-JS/)

Drag and drop feature is still in progress, not working at the moment. There is a commented piece of code I will remove in the next commit, but I need it for now.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -18,6 +19,12 @@ sort(tasks, 'index');
 
 let active = false;
 
+const remm = (index) => {
```
Comment (bellom) [inline src/index.js:22]: ### JavaScript requirements

Let's improve our naming conversion. `remm` , `rem`, `revs` let's use something readable.
checkout naming from [JS best practice](https://devinduct.com/blogpost/22/javascript-clean-code-best-practices)

image

File: src/index.js [hunk 2]
```diff
@@ -46,10 +76,11 @@ const content = (arr) => {
         li.style.top = `${e.clientY - 26}px`;
       }
     });
-
-    drag.addEventListener('mouseup', () => {
+    drag.addEventListener('mouseup', (e) => {
       li.classList.remove('dragging');
       li.style.position = 'inherit';
+      console.log(`You targeted ${e.target}`);
+      console.log(e.target);
```
Comment (bellom) [inline src/index.js:83]: ### JavaScript best practices

Kindly remove `console.log`

image

File: src/index.js [hunk 3]
```diff
@@ -46,10 +76,11 @@ const content = (arr) => {
...
       active = false;
     });
```
Comment (bellom) [inline src/index.js:86]: `Line 93`

`clear.addEventListener('click', () => {`

While edit and clear completed tasks seem working 👍  but we need to create a function just the way you create a function for `add` and `remove`.

image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @ReshyResh  👏 , 

### Awesome job so far!! :+1: 

#### You have done wonderfully well setting up linter and good commit messages 🔥, However, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Required Changes! ♻️ 

_1. Kindly check the comments under the review._

_2. Kindly remove all hardcoded items in the array:
image_

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ReshyResh** [comment]
There is no hard-coded array. The content.js is just the default content that is loaded in case there is no local storage in the browser, which makes the website look better in my opinion. I could remove it right away and it would't be an issue.

File: src/index.js (hunk 2 above)
Comment (rloterh) [reply src/index.js]: This issue is well resolved! ✔️

File: src/index.js [hunk 4]
```diff
@@ -34,6 +41,29 @@ const content = (arr) => {
...
     li.append(`${arr[i].description}`);
+
+    li.addEventListener('dblclick', () => {
+      li.classList.add('editing');
+      li.innerHTML = `<input type="checkbox" id="${i}"></input><input id = "change${i}" class="change" type = "text" value = "${arr[i].description}"></input><i id="remm${i}" class="drag fas fa-trash-alt"></i>`;
+      document.getElementById(`change${i}`).focus();
+      // document.getElementById(`change${i}`).addEventListener('keydown', () => {
+      //   arr[i].description = document.getElementById(`change${i}`).value;
+      //   setStorage(arr);
+      // });
```
[the remark this answers is not in the document]
Comment (rloterh) [reply src/index.js]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Robert, 

Your project is complete! Great job for making the necessary changes recommended by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾 🍾 :shipit: .
Congratulations! 🎉 🎉 🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**